glean-core/python/glean/net/http_client.py

```diff
@@ -60,10 +60,13 @@ def upload(cls, url: str, data: str, headers: List[Tuple[str, str]]) -> bool:
             )
             response = conn.getresponse()
         except http.client.HTTPException as e:
-            log.error("http.client.HTTPException: {}".format(e))
+            log.error("http.client.HTTPException: '{}' {}".format(url, e))
             return False
         except socket.gaierror as e:
-            log.error("socket.gaierror: {}".format(e))
+            log.error("socket.gaierror: '{}' {}".format(url, e))
+            return False
+        except OSError as e:
```

Comment:
Strictly speaking this is a separate bug -- we get this error if DNS lookup fails.  In typical Python fashion the exceptions that `conn.getresponse` can throw aren't documented, so we're playing whack-a-mole here.

Reply:
Since we don't really want this to ever throw, is there any value in just changing this to `except Exception as e:`?

Reply:
Yeah, I suppose that's fine.  Makes me feel icky in the event that anything truly unexpected *does* happen, but I should probably just get over that.